rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -517,10 +522,7 @@ void MasterGraph::output_routine()
                     }
                     _meta_data_graph->process(_augmented_meta_data);
                 }
-                if (full_batch_meta_data)
-                    full_batch_meta_data->concatenate(_augmented_meta_data);
-                else
-                    full_batch_meta_data = _augmented_meta_data->clone();
+                full_batch_meta_data = _augmented_meta_data->clone();
```

Comment:
use augmented_meta_data directly here. Why do you need to clone?

Reply:
The _augmented_meta_data is a shared pointer which is used by the reader.
Each time the metadata reader updates in the _augmented_metadata.
The _augmented_metadata will be overwritten for every batch hence we will not get the correct outputs as it may be overwritten by next batch hence we do a clone and move it to the meta ring buffer.

Reply:
Let us discuss it over a call